public/app/features/dashboard/dashgrid/PanelChrome.tsx

```diff
@@ -226,7 +250,13 @@ export class PanelChrome extends PureComponent<Props, State> {
           if (width === 0) {
             return null;
           }
+          if (this.queryWidthPixels < 0 && this.wantsQueryExecution) {
+            this.queryWidthPixels = width;
+            setTimeout(() => this.onRefresh(), 1); // TODO some better way!!!
+            return null;
```

Comment:
😢 there must be a better way!  @torkelo?

Reply:
onRefresh should definitely not be called here as it calls `setState`. I think if we changed the architecture and extract the querying logic into a separate component, then you could control the the refresh by two props, one to actually force the refresh and the second one to handle the side effect (via onRefresh prop).

But as you mentioned in your comment, extracting this abstraction is not on you next steps shortlist, and I would leave the setTimeout here until we extract that.